pygments/lexers/q.py

```diff
@@ -0,0 +1,228 @@
+"""
+    pygments.lexers.q
+    ~~~~~~~~~~~~~~~~~
+
+    Lexer for the Q programming language.
+
+    :copyright: Copyright 2006-2022 by the Pygments team, see AUTHORS.
+    :license: BSD, see LICENSE for details.
+"""
+
+from pygments.lexer import RegexLexer, words, include, bygroups
+from pygments.token import (
+    Comment,
+    Keyword,
+    Name,
+    Number,
+    Operator,
+    Punctuation,
+    String,
+    Text,
+    Whitespace,
+    Literal,
+    Generic,
+)
+
+__all__ = ["KLexer", "QLexer"]
+
+
+class KLexer(RegexLexer):
+    """
+    For `K <http://kx.com/>`_ source code.
+
+    .. versionadded:: 2.12
+    """
+
+    name = "K"
+    aliases = ["k"]
+    filenames = ["*.k"]
+
+    tokens = {
+        "whitespace": [
+            # hashbang script
+            (r"^#!.*$", Comment.Hashbang),
+            # Comments
+            (r"^/\s*\n", Comment.Multiline, "comments"),
+            (r"[ \t]+/.*", Comment.Single),
```

Comment:
yes, the syntax of k is quite interesting.  the `/` character is in-fact an operator when it comes directly after any character other than white space.  the following, for example, can compute 10 factorial
```
  */1+!10
3628800
```
but adding a space turns everything after the `*` into a comment
```
  * / 100
*
```

Reply:
OK. On the other hand, `[ \t]` doesn't include all possible whitespace characters -- doesn't it work with, say carriage returns? I think it would be easiest to use just two rules for comments:

```python
(r"^/\s*\n", Comment.Multiline, "comments"),
(r"(?<=^|\s)/.*", Comment.Single),
```

Reply:
the proposed ``(?<=^|\s)`` throws the following exception:

```
Error: error when loading custom lexer: uncompilable regex '(?<=^|\\s)/.*' in state 'whitespace' of <class 'QLexer'>: look-behind requires fixed-width pattern
```
negating the look-behind seems to do the trick:
```python
            (r"(?<!\S)/.*", Comment.Single),
```